src/jit/emitxarch.cpp

```diff
@@ -2901,174 +2901,282 @@ void emitter::emitInsStoreLcl(instruction ins, emitAttr attr, GenTreeLclVarCommo
 
 regNumber emitter::emitInsBinary(instruction ins, emitAttr attr, GenTree* dst, GenTree* src)
 {
-    // dst can only be a reg or modrm
-    assert(!dst->isContained() || dst->isUsedFromMemory() || instrIs3opImul(ins)); // dst on these isn't really the dst
+    // We can only have one memory operand and only src can be a constant operand
+    GenTree* memOp = nullptr;
+    GenTree* cnsOp = nullptr;
+    GenTree* otrOp = nullptr;
 
-#ifdef DEBUG
-    // src can be anything but both src and dst cannot be addr modes
-    // or at least cannot be contained addr modes
-    if (dst->isUsedFromMemory())
+    if (dst->isContained() || (dst->isLclField() && (dst->gtRegNum == REG_NA)) || dst->isUsedFromSpillTemp())
```

Comment:
I wasn't sure why `dst` isn't marked as contained for `(dst->isLclField() && (dst->gtRegNum == REG_NA))` (see [here](https://github.com/dotnet/coreclr/pull/15836/files#diff-a51165bc1fd2c0fa3987370e2e714356L2959) for the old code), it seems to map to a `GT_STORE_LCL_FLD` and, being a store, I would have expected it to be containable...

Reply:
Hmm, a contained `dst` is like rare in this code path. This might worth a separate investigation to avoid surprises.